test/Test.re

```diff
@@ -448,46 +463,41 @@ let suite =
         GlobalState.reset();
         let box_ = <BoxWithDynamicKeys title="box to move" />;
         let rendered0 = RenderedElement.render(box_);
-        let (rendered1, _) =
+        TestRenderer.convertElement(rendered0)
+        |> check(
+             renderedElement,
+             "Initial Box",
+             [TestComponents.(<BoxWithDynamicKeys id=1 state="box to move" />)]
+           );
+        let (rendered1, _) as actual1 =
           RenderedElement.update(
             rendered0,
             Nested(
               "div",
               [ReasonReact.stringToElement("before"), Nested("div", [box_])]
```

Comment:
We need to assert that the rendered `box_` instances here are physically equal (`===`) for `rendered0` and `rendered1`